arm-machinelearning/2016-05-01-preview/swagger/commitmentPlans.json

```diff
@@ -0,0 +1,903 @@
+                "location": {
+                    "description": "Resource Location",
+                    "type": "string"
+                },
+                "type": {
+                    "description": "Resource type",
+                    "type": "string",
+                    "readOnly": true
+                },
+                "tags": {
+                    "description": "Resource tags",
+                    "type": "object",
+                    "additionalProperties": {
+                        "type": "string"
+                    }
+                }
+            }
+        },
+        "CommitmentAssociation": {
+            "type": "object",
+            "description": "Represents the association between a commitment plan and some other resource, such as a Machine Learning web service.",
+            "properties": {
+                "etag": {
+                    "type": "string",
+                    "description": "An entity tag used to enforce optimistic concurrency."
+                },
+                "id": {
+                    "type": "string",
+                    "description": "The commitment association ARM resource ID."
+                },
+                "location": {
+                    "type": "string",
+                    "description": "The resource location."
+                },
+                "name": {
+                    "type": "string",
+                    "description": "The resource name.",
+                    "readOnly": true
+                },
+                "properties": {
+                    "$ref": "#/definitions/CommitmentAssociationProperties"
+                },
+                "type": {
+                    "type": "string",
+                    "description": "The type of the resource: Microsoft.MachineLearning/commitmentPlans/commitmentAssociations",
+                    "readOnly": true
+                }
+            }
+        },
+        "CommitmentAssociationProperties": {
+            "type": "object",
+            "description": "Properties of a commitment association resource.",
+            "properties": {
+                "associatedResourceId": {
+                    "type": "string",
+                    "description": "The ID of the resource this association points to, such as the ARM ID of a Machine Learning web service."
+                },
+                "commitmentPlanId": {
+                    "type": "string",
+                    "description": "The ARM ID of the parent commitment plan resource."
+                },
+                "creationDate": {
+                    "format": "date-time",
+                    "type": "string",
+                    "description": "The date at which this commitment association was created."
+                }
+            }
+        },
+        "ResourceSku": {
+            "type": "object",
+            "description": "The SKU of a resource.",
+            "properties": {
+                "capacity": {
```

Comment:
Are there a list of supported capacity types. Can the user put 10000? If the list of supported capacity values are relatively small then it would be better to describe it as an enum. If it is a range then provide the min and max constraint. This will help the customer tremendously.